packages/plugins/other/visitor-plugin-common/src/base-resolvers-visitor.ts

```diff
@@ -819,15 +788,15 @@ export class BaseResolversVisitor<
       }
 
       if (!isMapped && hasDefaultMapper && hasPlaceholder(this.config.defaultMapper.type)) {
-        // Make sure the inner type has no ResolverTypeWrapper
-        const name = clearWrapper(isScalar ? this._getScalar(typeName) : prev[typeName]);
-        const replaced = replacePlaceholder(this.config.defaultMapper.type, name);
+        const originalTypeName = isScalar ? this._getScalar(typeName) : prev[typeName];
 
-        // Don't wrap Union with ResolverTypeWrapper, each inner type already has it
         if (isUnionType(schemaType)) {
-          prev[typeName] = replaced;
+          // Don't clear ResolverTypeWrapper from Unions
+          prev[typeName] = replacePlaceholder(this.config.defaultMapper.type, originalTypeName);
         } else {
-          prev[typeName] = applyWrapper(replacePlaceholder(this.config.defaultMapper.type, name));
+          const name = clearWrapper(originalTypeName);
+          const replaced = replacePlaceholder(this.config.defaultMapper.type, name);
+          prev[typeName] = applyWrapper(replaced);
```

Comment:
Previous logic suggests that it's possible for a union to be in the scalar map 🤔 I'm unsure when/if it's possible. 

However, the updated logic should be backward compatible as it generates the same partial and wrapper order.